test/server/http_dynamic_delay_filter_integration_test.cc

```diff
@@ -22,134 +24,75 @@ namespace Nighthawk {
-    cleanupUpstreamAndDownstream();
-    codec_client_ = makeHttpConnection(lookupPort("http"));
-    Envoy::IntegrationStreamDecoderPtr response;
-    if (setup_for_upstream_request) {
-      response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
-    } else {
-      response = codec_client_->makeHeaderOnlyRequest(request_headers);
-      response->waitForEndStream();
-    }
-    return response;
-  }
-
-  const Envoy::Http::TestRequestHeaderMapImpl request_headers_;
-  const Envoy::Http::LowerCaseString delay_header_string_;
+public:
+  HttpDynamicDelayIntegrationTest() : HttpFilterIntegrationTestBase(GetParam()){};
 };
 
 INSTANTIATE_TEST_SUITE_P(IpVersions, HttpDynamicDelayIntegrationTest,
                          testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()));
 
 // Verify expectations with an empty dynamic-delay configuration.
 TEST_P(HttpDynamicDelayIntegrationTest, NoStaticConfiguration) {
-  setup(R"(
+  initializeFilterConfiguration(R"(
 name: dynamic-delay
 typed_config:
   "@type": type.googleapis.com/nighthawk.server.ResponseOptions
 )");
   // Don't send any config request header
-  getResponse();
-  EXPECT_EQ(upstream_request_->headers().get(delay_header_string_), nullptr);
+  getResponse(ResponseOrigin::UPSTREAM);
+  EXPECT_EQ(upstream_request_->headers().get(kDelayHeaderString), nullptr);
   // Send a config request header with an empty / default config. Should be a no-op.
-  getResponse("{}");
-  EXPECT_EQ(upstream_request_->headers().get(delay_header_string_), nullptr);
-  // Send a config request header, this should become effective.
-  getResponse("{static_delay: \"1.6s\"}");
-  EXPECT_EQ(upstream_request_->headers().get(delay_header_string_)->value().getStringView(),
-            "1600");
 
-  // Send a malformed config request header. This ought to shortcut and directly reply,
-  // hence we don't expect an upstream request.
-  auto response = getResponse("bad_json", false);
-  EXPECT_EQ(Envoy::Http::Utility::getResponseStatus(response->headers()), 500);
-  EXPECT_EQ(
-      response->body(),
-      "dynamic-delay didn't understand the request: Error merging json config: Unable to parse "
-      "JSON as proto (INVALID_ARGUMENT:Unexpected token.\nbad_json\n^): bad_json");
-  // Send an empty config header, which ought to trigger failure mode as well.
-  response = getResponse("", false);
-  EXPECT_EQ(Envoy::Http::Utility::getResponseStatus(response->headers()), 500);
-  EXPECT_EQ(
-      response->body(),
-      "dynamic-delay didn't understand the request: Error merging json config: Unable to "
-      "parse JSON as proto (INVALID_ARGUMENT:Unexpected end of string. Expected a value.\n\n^): ");
+  setRequestLevelConfiguration("{}");
+  getResponse(ResponseOrigin::UPSTREAM);
+  EXPECT_EQ(upstream_request_->headers().get(kDelayHeaderString), nullptr);
+  // Send a config request header, this should become effective.
```

Comment:
Rather than saying this should become effective, can we describe what the behavior should be?